### ISSUE TITLE
arcfaceloss+regularfaceregulariser yields negative loss value

### ISSUE THREAD

**Issue by jamesrobertwilliams** [KevinMusgrave/pytorch-metric-learning#443]
I am trying to do metric learning on a two class problem(I know!) using arcfaceloss and regularface regulariser and I see that after an epoch or so, this loss goes to negative. Is this like a known issue? Almost seems like the classes are back to back. When I rrt using different scale and margin, it seems to make no difference :(

Without the regulaiser, everything is fine and I get `good` results.

Any ppinters would be great.

**Comment by KevinMusgrave** [comment]
I haven't tried this combination. Negative loss isn't necessarily incorrect. Is the accuracy bad?

**Comment by KevinMusgrave** [comment]
The authors of the RegularFace paper say it doesn't necessarily improve on ArcFace. See https://kaizhao.net/regularface

**Comment by jamesrobertwilliams** [comment]
Thats very interesting - I stopped training bevcause I encountered negative loss..
Ill try read the paper again. For the problems I tried, this regulariser provided superior results in my use cases (classification).
